src/cli/lint/internalLints/upgrade/patches/8.2.1/collapseDuplicateMergeKeys.ts

```diff
@@ -0,0 +1,82 @@
+import { inspect } from 'util';
+
+import { glob } from 'fast-glob';
+import { promises as fs } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { log } from '../../../../../../utils/logging';
+
+const collapseDuplicateMergeKeys: PatchFunction = async ({
+  mode,
+}): Promise<PatchReturnType> => {
+  const buildkiteFiles = await glob(
+    ['.buildkite/**/*.yml', '.buildkite/**/*.yaml'],
+    { onlyFiles: true },
+  );
+
+  if (buildkiteFiles.length === 0) {
+    return { result: 'skip', reason: 'no Buildkite files found' };
+  }
+
+  const input = await Promise.all(
+    buildkiteFiles.map((name) => fs.readFile(name, 'utf-8')),
+  );
+
+  const replaced = input.map(collapseDuplicateMergeKeysInFile);
+
+  if (replaced.every((r, i) => r === input[i])) {
+    return { result: 'skip', reason: 'no duplicate merge keys found' };
+  }
+
+  if (mode === 'lint') {
+    return { result: 'apply' };
+  }
+
+  await Promise.all(
+    buildkiteFiles
+      .map((name, i) => [name, i] as const)
+      .filter(([, i]) => replaced[i] !== input[i])
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      .map(([name, i]) => fs.writeFile(name, replaced[i]!)),
```

Comment:
```suggestion
  buildkiteFiles.flatMap((name, i) =>
    replaced[i] !== input[i] ? [fs.writeFile(name, replaced[i]!)] : []
  ),
```

Is the first map doing much?

Reply:
Yeah the first map is so we keep `i` after filtering but yours does that too!